### PR TITLE
Initial package creation

### PR DESCRIPTION
**Pull request by kaderhashmi**

### CODE REVIEW

File: Dfc.Session/SessionClient.cs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                {
+                    Secure = true,
+                    IsEssential = true,
+                    HttpOnly = true,
+                    SameSite = SameSiteMode.Strict,
+                });
+            }
+            else
+            {
+                logger?.LogWarning($"SessionId not valid for session '{userSession.SessionId}'");
```
Comment (Muthuramana) [inline Dfc.Session/SessionClient.cs]: We need to through ArgumentException at this point.

File: Dfc.Session/SessionClient.cs [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            string sessionId = null;
+
+            var cookieSessionId = request.Cookies[SessionName];
+            if (!string.IsNullOrWhiteSpace(cookieSessionId))
+            {
+                sessionId = cookieSessionId;
+            }
+
+            var queryDictionary = System.Web.HttpUtility.ParseQueryString(request.QueryString.ToString());
+            var queryStringSessionId = queryDictionary.Get("sessionId");
```
Comment (Muthuramana) [inline Dfc.Session/SessionClient.cs]: change this to "dfc-session" please

File: Dfc.Session/SessionClient.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+            if (!string.IsNullOrWhiteSpace(queryStringSessionId))
+            {
+                sessionId = queryStringSessionId;
+            }
+
+            if (request.HasFormContentType)
+            {
+                var formData = await request.ReadFormAsync().ConfigureAwait(false);
+                var formSessionId = GetFormValue("sessionId", formData);
```
Comment (Muthuramana) [inline Dfc.Session/SessionClient.cs]: change this to "dfc-session" please

File: README.md [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+| DFC.Session.Package | appsettings-template.json | appsettings.json |
+
+## Configuring to run locally
+
+The project contains a number of "appsettings-template.json" files which contain sample appsettings for the web app and the integration test projects. To use these files, rename them to "appsettings.json" and edit and replace the configuration item values with values suitable for your environment.
+
+
+## Deployments
+
+This package can be used as part of a larger solution that need to search for courses.
```
Comment (Muthuramana) [inline README.md]: Think this readme needs updating...

File: Dfc.Session/SessionClient.cs (hunk 1 above)
Comment (kaderhashmi) [reply Dfc.Session/SessionClient.cs]: Updated as suggested

File: Dfc.Session/SessionClient.cs (hunk 3 above)
Comment (kaderhashmi) [reply Dfc.Session/SessionClient.cs]: Updated as suggested

File: Dfc.Session/SessionClient.cs (hunk 2 above)
Comment (kaderhashmi) [reply Dfc.Session/SessionClient.cs]: Updated as suggested

File: README.md (hunk 4 above)
Comment (kaderhashmi) [reply README.md]: Updated readme

**Pull request merged**